Dev/dayt/js server

daytondep opened the pull request:
Adding some more functionality to server code. As of right now the code shows database output when you travel to https://coursechart.herokuapp.com/test_table

The next step would be to add a button to allow us to navigate to that page without manually entering the url, and also properly filling some of the db.

Let me know if this implementation raises any glaring issues.

daytondep committed:
Preliminary server code (WIP)

daytondep committed:
Prelminary server code (#2)

daytondep committed:
Fix server error

daytondep committed:
Fix Server Code [2]

jkirkwin commented on server/queries.js:
```diff
@@ -0,0 +1,30 @@
+const { Client } = require('pg');
```
might want a `use strict;` at the top of this file - we're using it in `index.js` for code quality so might as well here too

jkirkwin commented on server/index.js:
```diff
@@ -18,6 +18,7 @@ client.connect();
 
 const express = require('express');
 const app = express();
+const db = require('./queries');
```
I dig putting all the db interaction stuff in a separate place from the routing logic 👍 the calls look clean

jkirkwin commented on server/queries.js:
```diff
@@ -0,0 +1,30 @@
+
+client.connect();
+
+const getTestTable = (req,res) => {
```
Can we nuke the comments in here? Can't tell if they're stale code or something else

jkirkwin commented on server/queries.js:
```diff
@@ -0,0 +1,30 @@
+const { Client } = require('pg');
+
+const client = new Client({
```
My only gripe here is that now we're using two connections to the db, one in this file and one in `index.js`. Not a huge deal now since we're not rly doing anything yet, but might be better to move that temporary logging stuff that's in `index.js` in here so we can re-use a single connection object anyway. Two is NBD but we do have some limit (I think 20?) on the number of connections we can have.

daytondep committed:
Added "use strict"

daytondep committed:
Moved all databaste code to queries.js

Moved all database code to queries.js, connection is now only
established in quesies.js rather than both index and queries.js, npm
test changed to test all files in server directory.
